perun-core/src/main/java/cz/metacentrum/perun/core/provisioning/HierarchicalHashedDataGenerator.java

```diff
@@ -106,7 +107,7 @@ private GenResourceDataNode getDataForResource(Resource resource) {
 		} else {
 			members = sess.getPerunBl().getResourcesManagerBl().getAllowedMembers(sess, resource);
 		}
-		if (BeansUtils.getCoreConfig().getForceConsents()) {
+		if (consentEval && BeansUtils.getCoreConfig().getForceConsents()) {
```

Comment:
we'll want to filter out members that do not have consent even if calling it without the consentEval flag, we just want to skip creation of new consents in that case